pixels.go

```diff
@@ -0,0 +1,98 @@
+package waifu2x
+
+import (
+	"fmt"
+	"math"
+)
+
+type Pixels struct {
+	Width  int
+	Height int
+	Pix    []uint8 // 0以下を0, 255以上を255 とする
+}
+
+func NewPixels(width, height int) *Pixels {
+	return &Pixels{
+		Width:  width,
+		Height: height,
+		Pix:    make([]uint8, width*height),
+	}
+}
+
+func (p Pixels) NewImagePlane() *ImagePlane {
+	pl := NewImagePlane(p.Width, p.Height)
+	for i := 0; i < len(p.Pix); i++ {
+		pl.Pix[i] = float64(p.Pix[i]) / 255.0
+	}
+	return pl
+}
+
+func (p Pixels) Decompose() (r, g, b, a *Pixels, err error) {
+	if len(p.Pix) != p.Width*p.Height*4 {
+		const msg = `decompose error, widht:%d, height:%d, buf len:%d`
+		err = fmt.Errorf(msg, p.Width, p.Height, len(p.Pix))
+		return
+	}
+	r = NewPixels(p.Width, p.Height)
+	g = NewPixels(p.Width, p.Height)
+	b = NewPixels(p.Width, p.Height)
+	a = NewPixels(p.Width, p.Height)
+	for w := 0; w < p.Width; w++ {
+		for h := 0; h < p.Height; h++ {
+			i := w + h*p.Width
+			r.Pix[i] = p.Pix[4*(w+h*p.Width)]
+			g.Pix[i] = p.Pix[4*(w+h*p.Width)+1]
+			b.Pix[i] = p.Pix[4*(w+h*p.Width)+2]
+			a.Pix[i] = p.Pix[4*(w+h*p.Width)+3]
+		}
+	}
+	return
+}
+
+func Compose(r, g, b, a *Pixels) (*Pixels, error) {
```

Comment:
__[golint]__
* [golint] exported function Compose should have comment or be unexported